whylog/assistant/regex_assistant/regex_match.py

```diff
@@ -57,17 +57,21 @@ def update_by_regex(self, new_regex):
         )
         self.regex = new_regex
 
-    def update_by_regex_object(self, regex_object):
-        self.regex = regex_object.pattern
-        self.param_groups = regex_object.param_groups
+    def update_by_pattern_match(self, pattern_match):
+        self.regex = pattern_match.pattern
+        self.param_groups = pattern_match.param_groups
 
     def update_by_guessed_regex(self, regex_id):
-        self.update_by_regex_object(self.guessed_regex_objects[regex_id])
+        self.update_by_pattern_match(self.guessed_pattern_matches[regex_id])
 
     def _guess_regexes(self):
-        guessed_objects = guess_regex_objects(self.line_text)
-        guessed_dict = dict([(key, guessed_objects[key]) for key in range(len(guessed_objects))])
-        self.guessed_regex_objects = guessed_dict
+        guessed_pattern_matches = guess_pattern_match(self.line_text)
+        guessed_dict = dict(
```

Comment:
what exactly happens here? dict is rewritten into a dict which looks exactly the same? why?